app/src/main/java/com/example/newsportal/app/bookmarks/BookmarksAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package com.example.newsportal.app.bookmarks
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.newsportal.R
+import com.example.newsportal.app.bookmarks.touchhelper.ItemTouchHelperAdapter
+import com.example.newsportal.app.topnews.NewsDiffCallback
+import com.example.newsportal.databinding.ListItemBinding
+import com.example.newsportal.domain.model.Article
+
+class BookmarksAdapter(
+    val onItemClick: (Article) -> Unit,
+    val onItemSwiped: (Article) -> Unit
+) : ListAdapter<Article, BookmarksAdapter.ItemViewHolder>(NewsDiffCallback),
+    ItemTouchHelperAdapter {
+
+    inner class ItemViewHolder(private val binding: ListItemBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(article: Article) {
+            with(article) {
+                binding.btnBookmark.visibility = View.GONE
+                binding.itemTitle.text = title
+                Glide.with(binding.root).load(urlToImage).error(R.drawable.ic_newspaper2)
+                    .centerCrop().into(binding.itemImage)
+                binding.itemTitle.setOnClickListener { onItemClick(this) }
```

Comment:
Typically, the user expects that when clicking anywhere on the card, a click will occur.

Reply:
You can use itemView for this.